developing/nix/README.md

```diff
@@ -77,48 +77,211 @@ docker and a native development environment.
+
+#### Installing New Packages
+
+```shell
+nix-env -i <package>
+```
+
+E.g.
+
+```shell
+nix-env -i direnv
+```
+
+#### Seeing Installed Packages
+
+```shell
+nix-env -q
+```
+
+#### Looking for Packages
+
+```shell
+nix-env -qa <package name>
+```
 
-    nix-env -p /nix/var/nix/profiles/myproject -f ./nix -i
+E.g.
+
+```shell
+nix-env -qa python3
+```
 
-Create an `.envrc` file in a directory with some environment variables your project needs:
+#### Uninstalling a Package
 
+```shell
+nix-env -e <package>
 ```
+
+E.g.
+
+```shell
+nix-env -e chamber
+```
+
+### Profiles
+
+Profiles can be used at a global level, on a per-project basis, or
+on a per-repo basis.
+
+E.g. for MilMove, you might want a profile that contains things like
+`aws-vault`, `chamber`, etc. which spans more than just the `mymove`
+repo. On the other hand, you may not want to use a global profile
+(like the default one mentioned in the next section) because you may
+need different versions of `aws-vault` for different projects.
+
+#### Default
+
+When it's installed, `nix` creates a profile which will store all the
+packages you install by default, unless you tell it to use a different
+profile when you're installing a package, or you switch the active
+profile to be another one.
+
+Default profile is in: `/nix/var/nix/profiles/per-user/<username>/profile/`
+E.g. `/nix/var/nix/profiles/per-user/felipe/profile/`
+
+#### Creating a Profile
+
+You can create a new profile wherever you want, though convention seems
+to be to do so in
+`/nix/var/nix/profiles/`
+or
+`/nix/var/nix/profiles/per-user/<username>/`
+
+To create a new profile, you can run the command below. Note that this
+won't actually set this new profile as the active one, that's in the
+next section.
+
+```shell
+nix-env -p /nix/var/nix/profiles/<profile name> -i nix nss-cacert
+```
+
+E.g.
+
+```shell
+nix-env -p /nix/var/nix/profiles/milmove -i nix nss-cacert
+```
+
+The `-p` is telling it the path to the profile you want to use for
+this `nix` command. This can be used to work with profiles that aren't
+the active profile. It can also be used with profiles that don't yet
+exist (like here) or with exising profiles.
+
+You may remember the `-i` is used for installing, so we're installing
+`nix` and `nss-cacert` into our new profile.
+
+This is needed because of an issue if you just switch to a new profile.
+If you don't install `nix` itself, you lose access to `nix` commands.
+And without `nss-cacert`, you can't install packages because `nix` will
+get SSL cert errors.
+
+For more info see the
+[nix github issue](https://github.com/NixOS/nix/issues/1396)
+
+#### Switching Profiles
+
+To switch profiles, run
+
+```shell
+nix-env -S <path to profile>
+```
+
+E.g.
+
+```shell
+nix-env -S /nix/var/nix/profiles/milmove
+```
+
+Note that it is possible to switch to a profile that doesn't exist yet
+and break `nix` commands. See creating profiles section for more info.
```

Comment:
IMHO, this shouldn't be necessary. We should be using `NIX_PROFILE` in `.envrc` to switch profiles

Reply:
This is more if you want to manage profiles at a level outside of a repo. In repos, I think making `.envrc` take care of managing profiles is good.

Reply:
Again, for simplicity's sake, I'm not convinced we need folks worrying about switching repos and breaking things.